implementation.js

```diff
@@ -1,42 +1,107 @@
 'use strict';
 
-var isNaN = require('is-nan');
-
-var $isFinite = isFinite;
-var $Number = Number;
-var abs = Math.abs;
+module.exports = function fround(x) {
+	/*
+	 * We use magic numbers to guarantee immediate values.
+	 * We extensively document their value in accompanying comments.
+	 */
+	/* eslint-disable no-magic-numbers */
 
-var BINARY_32_EPSILON = 1.1920928955078125e-7; // 2 ** -23;
-var BINARY_32_MIN_VALUE = 1.1754943508222875e-38; // 2 ** -126;
-var BINARY_32_MAX_VALUE = 3.4028234663852886e+38; // 2 ** 128 - 2 ** 104
-var EPSILON = 2.220446049250313e-16; // Number.EPSILON
+	// Apply ToNumber(x) without relying on any function
+	var v = +x; /* eslint-disable-line no-implicit-coercion */
 
-var inverseEpsilon = 1 / EPSILON;
-var roundTiesToEven = function roundTiesToEven(n) {
-	// Even though this reduces down to `return n`, it takes advantage of built-in rounding.
-	return (n + inverseEpsilon) - inverseEpsilon;
-};
+	var s = v < 0 ? -1 : 1; // 1 for NaN, +0 and -0
+	var mod = s * v; // abs(v), or -0 if v is -0
 
-module.exports = function fround(x) {
-	var v = $Number(x);
-	if (v === 0 || isNaN(v) || !$isFinite(v)) {
-		return v;
-	}
-	var s = v < 0 ? -1 : 1;
-	var mod = abs(v);
-	if (mod < BINARY_32_MIN_VALUE) {
-		/* eslint operator-linebreak: [2, "before"] */
-		return (
-			s
-			* roundTiesToEven(mod / BINARY_32_MIN_VALUE / BINARY_32_EPSILON)
-			* BINARY_32_MIN_VALUE * BINARY_32_EPSILON
-		);
-	}
-	// Veltkamp's splitting (?)
-	var a = (1 + (BINARY_32_EPSILON / EPSILON)) * mod;
-	var result = a - (a - mod);
-	if (result > BINARY_32_MAX_VALUE || isNaN(result)) {
+	/*
+	 * Overflow case.
+	 *
+	 * The magic number 3.4028235677973366E38 is the threshold from which fround
+	 * causes an overflow, resulting in Infinity.
+	 * This code path is also used when the input is already an Infinity.
+	 */
+	if (mod >= 3.4028235677973366E38) { // overflow-threshold
```

Comment:
The old implementation used `3.4028234663852886e+38`, which is lower than this number. I have no idea about what is expected, but could you add tests for numbers around these values?

We already have one:
https://github.com/es-shims/Math.fround/blob/be55c4c233d19494399f866bfe90d8891cc4fbf7/test/tests.js#L48-L49

It would probably be good to test both the new magic number and `3.4028235677973362e+38`, which is the float that comes right before it.

Reply:
> The old implementation used `3.4028234663852886e+38`, which is lower than this number.

The old implementation used the max-binary32 value. The new implementation refers to the *midpoint* between max-binary32 and Infinity in binary32.


> It would probably be good to test both the new magic number and `3.4028235677973362e+38`, which is the float that comes right before it.

The new magic number was already tested in "returns infinity for large numbers". I have added a test for the float just below it. I have also added tests around the subnormal-normal boundary.